packages/paste-core/components/form-pill-group/src/FormPillButton.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react';
+import {Box, safelySpreadBoxProps} from '@twilio-paste/box';
+import {ErrorIcon} from '@twilio-paste/icons/esm/ErrorIcon';
+import {defaultHoverStyles, defaultBaseStyles, errorHoverStyles, errorBaseStyles} from './FormPill.styles';
+
+interface FormPillStylesProps {
+  variant?: 'error' | 'default';
+  selected?: boolean;
+  element?: string;
+  children: React.ReactNode;
+  onKeyDown?: (event: React.KeyboardEvent<HTMLButtonElement>) => void;
+  onClick?: () => void;
+  isHoverable?: boolean;
+  isDismissable?: boolean;
+  /* We can't call this `disabled` because it conflicts with the internal `CompositeItem disabled prop */
+  isDisabled?: boolean;
+}
+
+export const FormPillButton = React.forwardRef<HTMLElement, FormPillStylesProps>(
+  (
+    {
+      element = 'FORM_PILL',
+      selected = false,
+      variant = 'default',
+      isHoverable = false,
+      isDisabled = false,
+      isDismissable = false,
+      ...props
+    },
+    ref
+  ) => {
+    const styles = React.useMemo(() => {
+      const hasHoverStyles = isHoverable && !isDisabled;
+      if (variant === 'error') {
+        if (hasHoverStyles) {
+          return {...errorBaseStyles, ...errorHoverStyles};
+        } else {
+          return errorBaseStyles;
+        }
+      } else {
+        if (hasHoverStyles) {
+          return {...defaultBaseStyles, ...defaultHoverStyles};
+        } else {
+          return defaultBaseStyles;
+        }
+      }
+    }, [isHoverable, isDisabled, variant]);
+
+    return (
+      <Box
+        {...safelySpreadBoxProps(props)}
+        element={element}
+        ref={ref}
+        aria-selected={selected}
+        aria-disabled={isDisabled}
+        role="option"
+        type="button"
+        as="button"
+        position="relative"
+        borderRadius="borderRadiusPill"
+        borderWidth="borderWidth10"
+        borderColor="transparent"
+        cursor="default"
+        height="24px"
```

Comment:
I set a solid height here because this height needs to match the `PillCloseIcon.tsx`component height and width. If we can use a new sizing token here, like `sizePillHeight` or something, that might be best as it would render in `rem` units instead of `px` units. This would allow the component to resize correctly when resizing font size at the browser level.